packages/starlight/components/LanguageSelect.astro

```diff
@@ -46,4 +46,18 @@ function localizedPathname(locale: string | undefined): string {
 		}
 	}
 	customElements.define('starlight-lang-select', StarlightLanguageSelect);
+	window.addEventListener('pageshow', (event) => {
+		if (!event.persisted) return;
+		document
+			.querySelectorAll<HTMLSelectElement>('starlight-lang-select select')
+			.forEach((select) => {
```

Comment:
Rather than this query selector, would it maybe make sense to move the event listener inside the custom element where we already get the `<select>`?

It means duplicating the `pageshow` listener, but we never have these more than twice on a page, so it feels like it would be “cleaner” to encapsulate it there. Unless there are other lifecycle reasons not to do it like that.